.github/workflows/cd-api-infra.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy API Infra
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "infra/api/**"
+
+jobs:
+  build-repository:
+    name: Build Repository
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.7.1
+          terraform_wrapper: false
+
+      - name: Configure AWS credentials
+        uses: ./.github/actions/configure-aws-credentials
+        with:
+          app_name: analytics
+          environment: shared
+      - name: Build Repository
+        run: |
+          ./bin/terraform-init-and-apply.sh infra/api/build-repository shared
+
+  deploy-infra:
+    name: Deploy Infrastructure
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        directory: ["database", "service"]
```

Comment:
We already deploy the infra for `service` in API deployments. It starts in the GH Action `deploy.yml`:

https://github.com/HHS/simpler-grants-gov/blob/26376ced4a06680ffe207fcebd5db3c7381f947e/.github/workflows/deploy.yml#L42-L43

which calls `make release-deploy`:
https://github.com/HHS/simpler-grants-gov/blob/26376ced4a06680ffe207fcebd5db3c7381f947e/Makefile#L194-L197

that runs `deploy-release.sh`:
https://github.com/HHS/simpler-grants-gov/blob/26376ced4a06680ffe207fcebd5db3c7381f947e/bin/deploy-release.sh#L18

which calls back to the `Makefile` here and ends up doing a `terraform apply` on `infra/api/service`:
https://github.com/HHS/simpler-grants-gov/blob/26376ced4a06680ffe207fcebd5db3c7381f947e/Makefile#L111-L116

Reply:
Here's an example of what I'm talking about. I changed the `infra/api` folder, but because I didn't change `api/**` it didn't get deployed. This is non-ideal behavior.

https://github.com/HHS/simpler-grants-gov/pull/1759/